src/librustc_trans/mir/block.rs

```diff
@@ -372,11 +373,13 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
                 // is also constant, then we can produce a warning.
                 if const_cond == Some(!expected) {
                     if let Some(err) = const_err {
-                        let err = ConstEvalErr{ span: span, kind: err };
-                        let mut diag = bcx.tcx().sess.struct_span_warn(
-                            span, "this expression will panic at run-time");
-                        note_const_eval_err(bcx.tcx(), &err, span, "expression", &mut diag);
-                        diag.emit();
+                        if let Some(span) = self.diag_span(span, terminator.source_info.scope) {
+                            let err = ConstEvalErr{ span: span, kind: err };
```

Comment:
nit: put a space before `{`. (I know you just copy and pasted code from above, but might as well clean this up while you're here.)